core/jobs/src/main/java/com/cognifide/aet/job/common/utils/javaScript/JavaScriptJobExecutor.java

```diff
@@ -0,0 +1,60 @@
+/**
+ * AET
+ *
+ * Copyright (C) 2018 Cognifide Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.cognifide.aet.job.common.utils.javaScript;
+
+import org.openqa.selenium.WebDriver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JavaScriptJobExecutor {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(JavaScriptJobExecutor.class);
+
+  private final org.openqa.selenium.JavascriptExecutor executor;
+  private final String currentUrl;
+
+  public JavaScriptJobExecutor(WebDriver webDriver) {
+    executor = (org.openqa.selenium.JavascriptExecutor) webDriver;
+    currentUrl = webDriver.getCurrentUrl();
+  }
+
+  public JavaScriptJobResult execute(String jsSnippet, Object... elements) {
+    JavaScriptJobResult result;
+    try {
+      result = executeJs(jsSnippet, elements);
+    } catch (Exception ex) {
+      result = handleJsException(jsSnippet, ex);
+    }
+    return result;
+  }
+
+  private JavaScriptJobResult executeJs(String jsSnippet, Object... elements) {
+    if (LOGGER.isDebugEnabled()) {
+      LOGGER.debug("Executing JavaScript command: {} on page: {}", jsSnippet, currentUrl);
+    }
+    Object jsResult = executor.executeScript(jsSnippet, elements);
+    return new JavaScriptJobResult(jsResult);
+  }
+
+  private JavaScriptJobResult handleJsException(String jsSnippet, Exception ex) {
+    String message = String
+        .format("Can't execute JavaScript command. jsSnippet: \"%s\". Error: %s ",
+            jsSnippet, ex.getMessage());
+    LOGGER.warn(message, ex);
+    return new JavaScriptJobResult(ex);
```

Comment:
Wouldn't it be simpler to throw this exception here instead of forcing to call `isException` on the result obeject every time if we want to check if exception is thrown?

Reply:
Thanks for the good suggestion, code refactored!